middlewares/authenticate.js

```diff
@@ -0,0 +1,27 @@
+const User = require("../models/user.model");
+const jwt = require("jsonwebtoken");
+
+const authenticate = async (req, res, next) => {
+   try{
+    let token = req.headers.authorization;
+    if(!token){
+        return res.status(401).json({success:false, errorMessage:"Unauthorized. Token not passed."});
+    }
+    token = token.split(" ")[1];
+    const decodedValue = jwt.verify(token, process.env.JWT_SECRET);
+    const user = await User.findOne({_id: decodedValue._id});
+    if(!user){
+        return res.status(401).json({success:false, errorMessage:"Unauthorized. Either user is not registered or Token is invalid."});
+    }
+    user.__v = undefined;
+    user.password = undefined;
```

Comment:
You can directly store _id in req.user as
`req.user=user._id`
or
`req.user=decodedValue._id`
Because rest details are not required I think. Anyways check it out once :)